services/wallet/activity/service.go

```diff
@@ -316,14 +320,19 @@ func (s *Service) GetRecipientsAsync(requestID int32, chainIDs []w_common.ChainI
 		result.Addresses, result.HasMore, err = GetRecipients(ctx, s.db, chainIDs, addresses, offset, limit)
 		return result, err
 	}, func(result interface{}, taskType async.TaskType, err error) {
-		res := result.(*GetRecipientsResponse)
+		res := &GetRecipientsResponse{
+			ErrorCode: ErrorCodeFailed,
+		}
 		if errors.Is(err, context.Canceled) || errors.Is(err, async.ErrTaskOverwritten) {
 			res.ErrorCode = ErrorCodeTaskCanceled
-		} else if err != nil {
-			res.ErrorCode = ErrorCodeFailed
+		} else if err == nil {
+			recipientsResponse, ok := result.(*GetRecipientsResponse)
+			if ok {
+				res = recipientsResponse
+			}
```

Comment:
If `err == nil` and `result` can not be type asserted as `* GetRecipientsResponse` then this function will return :

```go
&GetRecipientsResponse{
			ErrorCode: ErrorCodeFailed,
		}
```

Reply:
that's fine, it's basically the same behavior you get from the other endpoints, since for some reason you don't have a valid result you can return.